src/SSHDebugPS/ProcFSOutputParser.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using System.Windows.Documents;
+
+namespace Microsoft.SSHDebugPS
+{
+    public class ProcFSOutputParser
+    {
+        public static string EscapedCommandText => CommandText.Replace("$", "\\$");
+        public static string CommandText => @"echo shell-process:$$; for filename in /proc/[0-9]*; do echo $filename,cmdline:$(tr '\0' ' ' < $filename/cmdline 2>/dev/null),ls:$(ls -lh $filename/exe 2>/dev/null); done";
```

Comment:
>(ls -lh $filename/exe 2>/dev/null) [](start = 176, length = 34)

Not saying this needs to change, but if you run into issues with parsing the output of "ls" on different platforms, you might try using "stat" to get the username instead, e.g.:

andrew@andrewcr-lnx-1:~$ stat -c %U /proc/70532/exe
andrew

Reply:
added a comment so it doesn't get lost